hermes/src/db/local_cache.rs

```diff
@@ -0,0 +1,88 @@
+use {
+    super::{
+        Db,
+        DbRecord,
+        RequestTime,
+    },
+    anyhow::Result,
+    dashmap::DashMap,
+    std::{
+        collections::VecDeque,
+        sync::Arc,
+    },
+};
+
+#[derive(Clone)]
+pub struct LocalCache {
+    cache:            Arc<DashMap<Vec<u8>, VecDeque<DbRecord>>>,
```

Comment:
In order to avoid the Sync problem we're relying on the fact the fields of these database objects are themselves Sync. This is fine but we should make sure to document that this is our design at the top of this file. If we were to introduce `sqlite` for example it would also require us to Arc<Mutex> the field and this should be clear.

Reply:
The Storage trait has the bounds on `Sync + Send` to ensure all implementations follow it.